File: core/models/models/metric.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Metric struct {
	any                  `collection:"metrics"`
	BaseModel            `bson:",inline"`
	Type                 string             `json:"type" bson:"type" description:"Type"`
	NodeId               primitive.ObjectID `json:"node_id" bson:"node_id" description:"Node ID"`
	CpuUsagePercent      float32            `json:"cpu_usage_percent" bson:"cpu_usage_percent" description:"CPU usage percentage"`
	TotalMemory          uint64             `json:"total_memory" bson:"total_memory" description:"Total memory"`
	AvailableMemory      uint64             `json:"available_memory" bson:"available_memory" description:"Available memory"`
	UsedMemory           uint64             `json:"used_memory" bson:"used_memory" description:"Used memory"`
	UsedMemoryPercent    float32            `json:"used_memory_percent" bson:"used_memory_percent" description:"Used memory percentage"`
	TotalDisk            uint64             `json:"total_disk" bson:"total_disk" description:"Total disk"`
	AvailableDisk        uint64             `json:"available_disk" bson:"available_disk" description:"Available disk"`
	UsedDisk             uint64             `json:"used_disk" bson:"used_disk" description:"Used disk"`
	UsedDiskPercent      float32            `json:"used_disk_percent" bson:"used_disk_percent" description:"Used disk percentage"`
	DiskReadBytesRate    float32            `json:"disk_read_bytes_rate" bson:"disk_read_bytes_rate" description:"Disk read bytes rate"`
	DiskWriteBytesRate   float32            `json:"disk_write_bytes_rate" bson:"disk_write_bytes_rate" description:"Disk write bytes rate"`
	NetworkBytesSentRate float32            `json:"network_bytes_sent_rate" bson:"network_bytes_sent_rate" description:"Network bytes sent rate"`
	NetworkBytesRecvRate float32            `json:"network_bytes_recv_rate" bson:"network_bytes_recv_rate" description:"Network bytes recv rate"`
}

File: frontend/crawlab-ui/src/i18n/lang/en/components/metric.ts
const metric: LComponentsMetric = {
  select: {
    placeholder: 'Select Metric',
  },
  metrics: {
    cpu_usage_percent: 'CPU Usage (%)',
    total_memory: 'Total Memory',
    available_memory: 'Available Memory',
    used_memory: 'Used Memory',
    used_memory_percent: 'Used Memory (%)',
    total_disk: 'Total Disk',
    available_disk: 'Available Disk',
    used_disk: 'Used Disk',
    used_disk_percent: 'Used Disk (%)',
    disk_read_bytes_rate: 'Disk Read IO',
    disk_write_bytes_rate: 'Disk Write IO',
    network_bytes_sent_rate: 'Network Sent IO',
    network_bytes_recv_rate: 'Network Recv IO',
    connections: 'Connections',
    query_per_second: 'Queries/sec',
    cache_hit_ratio: 'Cache Hit Ratio',
    replication_lag: 'Replication Lag',
    lock_wait_time: 'Lock Wait Time',
  },
  groups: {
    disk_io_bytes_rate: 'Disk IO',
    network_io_bytes_rate: 'Network IO',
  },
  timeUnits: {
    s: 'sec',
    m: 'min',
    h: 'hr',
    d: 'd',
    w: 'wk',
    M: 'mon',
    y: 'yr',
  },
  timeRanges: {
    '1h': 'Past 1 Hour',
    '24h': 'Past 24 Hours',
    '7d': 'Past 7 Days',
    '30d': 'Past 30 Days',
  },
  noData: {
    label: 'No Data',
    tooltip: 'No recent monitoring data available for this node',
  },
  unavailable: {
    chart: 'This metric is unavailable',
    option: 'Unavailable',
  },
};

export default metric;
